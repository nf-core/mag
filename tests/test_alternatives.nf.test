nextflow_pipeline {

    name "Test nf-core/mag"
    script "../main.nf"
    tag "pipeline"
    tag "test_alternatives"
    profile "test_alternatives"

    test("-profile test_alternatives") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {

            // Output dir: QC_shortreads
            def stable_name_qc = getAllFilesFromDir(
                params.outdir, include: ['QC_shortreads/**'], relative: true, includeDir: false
            )
            def stable_path_qc = getAllFilesFromDir(
                params.outdir, include: ['QC_shortreads/**'], ignoreFile: 'tests/.nftignore'
            )
            def remove_phix_logs = getAllFilesFromDir(
                params.outdir, include: ['QC_shortreads/remove_phix/*.log']
            )

            // Output dir: Assembly
            def stable_name_assembly = getAllFilesFromDir(
                params.outdir, include: ['Assembly/**'], relative: true, includeDir: false
            )
            def stable_path_assembly = getAllFilesFromDir(
                params.outdir, include: ['Assembly/**'], ignoreFile: 'tests/.nftignore'
            )
            def megahit_logs = getAllFilesFromDir(
                params.outdir, include: ['Assembly/MEGAHIT/*.log']
            )
            def bowtie2_logs = getAllFilesFromDir(
                params.outdir, include: ['Assembly/MEGAHIT/QC/*/*.bowtie2.log']
            )
            def megahit_contigs = getAllFilesFromDir(
                params.outdir, include: ['Assembly/MEGAHIT/*.contigs.fa.gz']
            )

            // Output dir: GenomeBinning
            def stable_name_binning = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/**'], relative: true, includeDir: false
            )
            def stable_path_binning = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/**'], ignoreFile: 'tests/.nftignore'
            )
            def checkm2_logs = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/QC/CheckM2/*/checkm2.log']
            )
            def checkm2_diamond_results = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/QC/CheckM2/**/DIAMOND_RESULTS.tsv']
            )
            def metabat2_bins = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/MetaBAT2/bins/*.fa.gz']
            )
            def metabat2_unbins = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/MetaBAT2/{discarded,unbinned}/*.fa.gz']
            )

            // Output dir: multiqc
            def stable_name_multiqc = getAllFilesFromDir(
                params.outdir, include: ['multiqc/**'], relative: true, includeDir: false
            )
            def stable_path_multiqc = getAllFilesFromDir(
                params.outdir, include: ['multiqc/**'], ignoreFile: 'tests/.nftignore'
            )

            assertAll(
                { assert workflow.success },
                { assert snapshot(
                        workflow.trace.succeeded().size(),
                        removeNextflowVersion("${outputDir}/pipeline_info/nf_core_mag_software_mqc_versions.yml"),
                    ).match()
                },
                { assert snapshot(stable_name_qc,       stable_path_qc      ).match('qc') },
                { assert snapshot(stable_name_assembly, stable_path_assembly).match('assembly') },
                { assert snapshot(stable_name_binning,  stable_path_binning ).match('binning') },
                { assert snapshot(stable_name_multiqc,  stable_path_multiqc ).match('multiqc') },
                { assert snapshot(
                        checkm2_diamond_results.collect { results -> results.text.readLines().size() },
                    ).match('checkm2_diamond_results')
                },
                // FASTA checks
                { assert megahit_contigs.collect { contigs -> path("${contigs}").fasta.size() > 0 }.every() },
                { assert metabat2_bins.collect { contigs -> path("${contigs}").fasta.size() > 0 }.every() },
                { assert metabat2_unbins.collect { contigs -> path("${contigs}").fasta.size() >= 0 }.every() },
                // Log checks
                { assert remove_phix_logs.collect {
                        log -> log.readLines().last() ==~ /[0-9.]+% overall alignment rate/
                    }.every()
                },
                { assert megahit_logs.collect {
                        log -> (log.readLines().last() =~ /ALL DONE. Time elapsed: [0-9.]+ seconds/).find()
                    }.every()
                },
                { assert bowtie2_logs.collect {
                        log -> log.readLines().last() ==~ /[0-9.]+% overall alignment rate/
                    }.every()
                },
                { assert checkm2_logs.collect {
                        log -> log.readLines().last().contains('CheckM2 finished successfully.')
                    }.every()
                },
            )
        }
    }
}
