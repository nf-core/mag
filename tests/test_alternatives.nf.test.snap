{
    "checkm2_diamond_results": {
        "content": [
            [
                806
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-21T06:18:59.114965285"
    },
    "-profile test_alternatives": {
        "content": [
            41,
            {
                "BIN_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "BOWTIE2_ASSEMBLY_ALIGN": {
                    "bowtie2": "2.4.2",
                    "samtools": 1.11,
                    "pigz": "2.3.4"
                },
                "BOWTIE2_ASSEMBLY_BUILD": {
                    "bowtie2": "2.4.2"
                },
                "BOWTIE2_PHIX_REMOVAL_ALIGN": {
                    "bowtie2": "2.4.2"
                },
                "BOWTIE2_PHIX_REMOVAL_BUILD": {
                    "bowtie2": "2.4.2"
                },
                "CHECKM2_DATABASEDOWNLOAD": {
                    "aria2": "1.37.0"
                },
                "CHECKM2_PREDICT": {
                    "checkm2": "1.1.0"
                },
                "CONCAT_BINQC_TSV": {
                    "csvtk": "0.31.0"
                },
                "DASTOOL_FASTATOCONTIG2BIN_TIARA": {
                    "dastool": "1.1.6"
                },
                "FASTQC_RAW": {
                    "fastqc": "0.12.1"
                },
                "FASTQC_TRIMMED": {
                    "fastqc": "0.12.1"
                },
                "GUNZIP_BINS": {
                    "gunzip": 1.13
                },
                "GUNZIP_SHORTREAD_ASSEMBLIES": {
                    "gunzip": 1.13
                },
                "MAG_DEPTHS": {
                    "python": "3.6.7",
                    "pandas": "1.1.5"
                },
                "MAG_DEPTHS_PLOT": {
                    "python": "3.9.6",
                    "pandas": "1.3.0",
                    "seaborn": "0.11.0"
                },
                "MAG_DEPTHS_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "MEGAHIT": {
                    "megahit": "1.2.9"
                },
                "METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS_SHORTREAD": {
                    "metabat2": 2.15
                },
                "METABAT2_METABAT2": {
                    "metabat2": 2.17
                },
                "SEQKIT_STATS": {
                    "seqkit": "2.9.0"
                },
                "SPLIT_FASTA": {
                    "python": "3.6.7",
                    "biopython": "1.7.4",
                    "pandas": "1.1.5"
                },
                "TIARA_SUMMARY": {
                    "csvtk": "0.31.0"
                },
                "TIARA_TIARA": {
                    "tiara": "1.0.3"
                },
                "TRIMMOMATIC": {
                    "trimmomatic": 0.39
                },
                "Workflow": {
                    "nf-core/mag": "v4.1.0dev"
                },
                "r-base": "4.1.3",
                "r-tidyverse": "1.3.1"
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-25T16:23:26.183903562"
    },
    "qc": {
        "content": [
            [
                "QC_shortreads/fastqc/test_minigut_run0_raw_1_fastqc.html",
                "QC_shortreads/fastqc/test_minigut_run0_raw_2_fastqc.html",
                "QC_shortreads/fastqc/test_minigut_run0_trimmed_1_fastqc.html",
                "QC_shortreads/fastqc/test_minigut_run0_trimmed_2_fastqc.html",
                "QC_shortreads/fastqc/test_minigut_sample2_run0_raw_1_fastqc.html",
                "QC_shortreads/fastqc/test_minigut_sample2_run0_raw_2_fastqc.html",
                "QC_shortreads/fastqc/test_minigut_sample2_run0_trimmed_1_fastqc.html",
                "QC_shortreads/fastqc/test_minigut_sample2_run0_trimmed_2_fastqc.html",
                "QC_shortreads/remove_phix/test_minigut_run0_phix_removed.bowtie2.log",
                "QC_shortreads/remove_phix/test_minigut_sample2_run0_phix_removed.bowtie2.log",
                "QC_shortreads/trimmomatic/test_minigut/test_minigut_run0_trimmomatic_trim.log",
                "QC_shortreads/trimmomatic/test_minigut_sample2/test_minigut_sample2_run0_trimmomatic_trim.log"
            ],
            [
                "test_minigut_run0_trimmomatic_trim.log:md5,1d755bf9117ce0af4c47b4ad0b358f53",
                "test_minigut_sample2_run0_trimmomatic_trim.log:md5,ea9f15adce3091c96ef7ba6285957676"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-21T06:18:58.788514681"
    },
    "assembly": {
        "content": [
            [
                "Assembly/MEGAHIT/MEGAHIT-test_minigut.contigs.fa.gz",
                "Assembly/MEGAHIT/MEGAHIT-test_minigut.log",
                "Assembly/MEGAHIT/MEGAHIT-test_minigut_sample2.contigs.fa.gz",
                "Assembly/MEGAHIT/MEGAHIT-test_minigut_sample2.log",
                "Assembly/MEGAHIT/QC/test_minigut/MEGAHIT-test_minigut-test_minigut_sample2.bowtie2.log",
                "Assembly/MEGAHIT/QC/test_minigut/MEGAHIT-test_minigut.bowtie2.log",
                "Assembly/MEGAHIT/QC/test_minigut_sample2/MEGAHIT-test_minigut_sample2-test_minigut.bowtie2.log",
                "Assembly/MEGAHIT/QC/test_minigut_sample2/MEGAHIT-test_minigut_sample2.bowtie2.log"
            ],
            [
                "MEGAHIT-test_minigut.contigs.fa.gz:md5,281287b2e105d82cc9d2900c7f624769",
                "MEGAHIT-test_minigut_sample2.contigs.fa.gz:md5,e88019689caed5802fe5596d94a38df8"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-21T06:18:58.837281134"
    },
    "binning": {
        "content": [
            [
                "GenomeBinning/MetaBAT2/bins/MEGAHIT-MetaBAT2-test_minigut.1.fa.gz",
                "GenomeBinning/MetaBAT2/bins/MEGAHIT-MetaBAT2-test_minigut.2.fa.gz",
                "GenomeBinning/MetaBAT2/discarded/MEGAHIT-MetaBAT2-test_minigut.lowDepth.fa.gz",
                "GenomeBinning/MetaBAT2/discarded/MEGAHIT-MetaBAT2-test_minigut.tooShort.fa.gz",
                "GenomeBinning/MetaBAT2/discarded/MEGAHIT-MetaBAT2-test_minigut_sample2.lowDepth.fa.gz",
                "GenomeBinning/MetaBAT2/discarded/MEGAHIT-MetaBAT2-test_minigut_sample2.tooShort.fa.gz",
                "GenomeBinning/MetaBAT2/unbinned/discarded/MEGAHIT-MetaBAT2-test_minigut.unbinned.pooled.fa.gz",
                "GenomeBinning/MetaBAT2/unbinned/discarded/MEGAHIT-MetaBAT2-test_minigut.unbinned.remaining.fa.gz",
                "GenomeBinning/MetaBAT2/unbinned/discarded/MEGAHIT-MetaBAT2-test_minigut_sample2.unbinned.pooled.fa.gz",
                "GenomeBinning/MetaBAT2/unbinned/discarded/MEGAHIT-MetaBAT2-test_minigut_sample2.unbinned.remaining.fa.gz",
                "GenomeBinning/QC/CheckM2/MEGAHIT-MetaBAT2-prokarya-unrefined-test_minigut/checkm2.log",
                "GenomeBinning/QC/CheckM2/MEGAHIT-MetaBAT2-prokarya-unrefined-test_minigut/diamond_output/DIAMOND_RESULTS.tsv",
                "GenomeBinning/QC/CheckM2/MEGAHIT-MetaBAT2-prokarya-unrefined-test_minigut/protein_files/MEGAHIT-MetaBAT2-test_minigut.1.faa",
                "GenomeBinning/QC/CheckM2/MEGAHIT-MetaBAT2-prokarya-unrefined-test_minigut/protein_files/MEGAHIT-MetaBAT2-test_minigut.2.faa",
                "GenomeBinning/QC/CheckM2/MEGAHIT-MetaBAT2-prokarya-unrefined-test_minigut/quality_report.tsv",
                "GenomeBinning/QC/CheckM2/MEGAHIT-MetaBAT2-prokarya-unrefined-test_minigut_checkm2_report.tsv",
                "GenomeBinning/QC/checkm2_summary.tsv",
                "GenomeBinning/Tiara/tiara_summary.tsv",
                "GenomeBinning/bin_summary.tsv",
                "GenomeBinning/depths/bins/MEGAHIT-MetaBAT2-test_minigut-binDepths.heatmap.png",
                "GenomeBinning/depths/bins/bin_depths_summary.tsv",
                "GenomeBinning/depths/contigs/MEGAHIT-test_minigut-depth.txt.gz",
                "GenomeBinning/depths/contigs/MEGAHIT-test_minigut_sample2-depth.txt.gz"
            ],
            [
                "MEGAHIT-MetaBAT2-test_minigut.1.fa.gz:md5,44b77f8e1a059011e45896cb80dbf361",
                "MEGAHIT-MetaBAT2-test_minigut.2.fa.gz:md5,10fb6fc82494beda78e6b1c6c2d4d8d2",
                "MEGAHIT-MetaBAT2-test_minigut.lowDepth.fa.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "MEGAHIT-MetaBAT2-test_minigut.tooShort.fa.gz:md5,4b65a9dd07c9b4bcdc828c104a7df36d",
                "MEGAHIT-MetaBAT2-test_minigut_sample2.lowDepth.fa.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "MEGAHIT-MetaBAT2-test_minigut_sample2.tooShort.fa.gz:md5,f9479e7fb34586a2892cd7bc323d414a",
                "MEGAHIT-MetaBAT2-test_minigut.unbinned.remaining.fa.gz:md5,7ac001a38113812061f8ef5c94e3f579",
                "MEGAHIT-MetaBAT2-test_minigut_sample2.unbinned.remaining.fa.gz:md5,e1a969d6e292e10a81ef4236907984a9",
                "MEGAHIT-MetaBAT2-test_minigut.1.faa:md5,ba436cbba64ce111f228da1e2daac642",
                "MEGAHIT-MetaBAT2-test_minigut.2.faa:md5,577adc1aa1ebd51fed46f726ad90d9b3",
                "quality_report.tsv:md5,9365828b072487af69333ecb35353255",
                "MEGAHIT-MetaBAT2-prokarya-unrefined-test_minigut_checkm2_report.tsv:md5,9365828b072487af69333ecb35353255",
                "checkm2_summary.tsv:md5,9365828b072487af69333ecb35353255",
                "tiara_summary.tsv:md5,6f97d6016921205c7913e54850b52dad",
                "bin_summary.tsv:md5,8901f5aee83fa37cfce832d60b51687e",
                "bin_depths_summary.tsv:md5,ef8b1a8aad59ad4f56515d0fb7045a6d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-21T06:18:58.895769923"
    },
    "multiqc": {
        "content": [
            [
                "multiqc/multiqc_data/BETA-multiqc.parquet",
                "multiqc/multiqc_data/bowtie2_pe_plot.yaml",
                "multiqc/multiqc_data/checkm2-first-table.yaml",
                "multiqc/multiqc_data/fastqc-1-status-check-heatmap.yaml",
                "multiqc/multiqc_data/fastqc-1_per_base_n_content_plot.yaml",
                "multiqc/multiqc_data/fastqc-1_per_base_sequence_quality_plot.yaml",
                "multiqc/multiqc_data/fastqc-1_per_sequence_gc_content_plot_Counts.yaml",
                "multiqc/multiqc_data/fastqc-1_per_sequence_gc_content_plot_Percentages.yaml",
                "multiqc/multiqc_data/fastqc-1_per_sequence_quality_scores_plot.yaml",
                "multiqc/multiqc_data/fastqc-1_sequence_counts_plot.yaml",
                "multiqc/multiqc_data/fastqc-1_sequence_duplication_levels_plot.yaml",
                "multiqc/multiqc_data/fastqc-status-check-heatmap.yaml",
                "multiqc/multiqc_data/fastqc_per_base_n_content_plot.yaml",
                "multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.yaml",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.yaml",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.yaml",
                "multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.yaml",
                "multiqc/multiqc_data/fastqc_sequence_counts_plot.yaml",
                "multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.yaml",
                "multiqc/multiqc_data/fastqc_sequence_length_distribution_plot.yaml",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_bowtie2.yaml",
                "multiqc/multiqc_data/multiqc_bowtie2_bowtie2-2.yaml",
                "multiqc/multiqc_data/multiqc_checkm2.yaml",
                "multiqc/multiqc_data/multiqc_citations.yaml",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_fastqc.yaml",
                "multiqc/multiqc_data/multiqc_fastqc_fastqc-1.yaml",
                "multiqc/multiqc_data/multiqc_general_stats.yaml",
                "multiqc/multiqc_data/multiqc_software_versions.yaml",
                "multiqc/multiqc_data/multiqc_sources.yaml",
                "multiqc/multiqc_data/multiqc_trimmomatic.yaml",
                "multiqc/multiqc_data/trimmomatic_plot.yaml",
                "multiqc/multiqc_plots/pdf/bowtie2_pe_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/bowtie2_pe_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/checkm2-first-table.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1-status-check-heatmap.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1_per_base_n_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1_per_base_sequence_quality_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1_per_sequence_gc_content_plot_Counts.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1_per_sequence_gc_content_plot_Percentages.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1_per_sequence_quality_scores_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1_sequence_counts_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1_sequence_counts_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-1_sequence_duplication_levels_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-status-check-heatmap.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_n_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_sequence_quality_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_quality_scores_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_duplication_levels_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_length_distribution_plot.pdf",
                "multiqc/multiqc_plots/pdf/trimmomatic_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/trimmomatic_plot-pct.pdf",
                "multiqc/multiqc_plots/png/bowtie2_pe_plot-cnt.png",
                "multiqc/multiqc_plots/png/bowtie2_pe_plot-pct.png",
                "multiqc/multiqc_plots/png/checkm2-first-table.png",
                "multiqc/multiqc_plots/png/fastqc-1-status-check-heatmap.png",
                "multiqc/multiqc_plots/png/fastqc-1_per_base_n_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc-1_per_base_sequence_quality_plot.png",
                "multiqc/multiqc_plots/png/fastqc-1_per_sequence_gc_content_plot_Counts.png",
                "multiqc/multiqc_plots/png/fastqc-1_per_sequence_gc_content_plot_Percentages.png",
                "multiqc/multiqc_plots/png/fastqc-1_per_sequence_quality_scores_plot.png",
                "multiqc/multiqc_plots/png/fastqc-1_sequence_counts_plot-cnt.png",
                "multiqc/multiqc_plots/png/fastqc-1_sequence_counts_plot-pct.png",
                "multiqc/multiqc_plots/png/fastqc-1_sequence_duplication_levels_plot.png",
                "multiqc/multiqc_plots/png/fastqc-status-check-heatmap.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_n_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_sequence_quality_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Counts.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Percentages.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_quality_scores_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-cnt.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-pct.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_duplication_levels_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_length_distribution_plot.png",
                "multiqc/multiqc_plots/png/trimmomatic_plot-cnt.png",
                "multiqc/multiqc_plots/png/trimmomatic_plot-pct.png",
                "multiqc/multiqc_plots/svg/bowtie2_pe_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/bowtie2_pe_plot-pct.svg",
                "multiqc/multiqc_plots/svg/checkm2-first-table.svg",
                "multiqc/multiqc_plots/svg/fastqc-1-status-check-heatmap.svg",
                "multiqc/multiqc_plots/svg/fastqc-1_per_base_n_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc-1_per_base_sequence_quality_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc-1_per_sequence_gc_content_plot_Counts.svg",
                "multiqc/multiqc_plots/svg/fastqc-1_per_sequence_gc_content_plot_Percentages.svg",
                "multiqc/multiqc_plots/svg/fastqc-1_per_sequence_quality_scores_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc-1_sequence_counts_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/fastqc-1_sequence_counts_plot-pct.svg",
                "multiqc/multiqc_plots/svg/fastqc-1_sequence_duplication_levels_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc-status-check-heatmap.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_n_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_sequence_quality_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Counts.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_quality_scores_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-pct.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_duplication_levels_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_length_distribution_plot.svg",
                "multiqc/multiqc_plots/svg/trimmomatic_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/trimmomatic_plot-pct.svg",
                "multiqc/multiqc_report.html"
            ],
            [
                "bowtie2_pe_plot.yaml:md5,17542811418a01c5c4497e77626bad91",
                "checkm2-first-table.yaml:md5,d1141b2bc74b8929807dcd93ece8143d",
                "fastqc-1-status-check-heatmap.yaml:md5,30f0c0bbefd3e9c549116c8b7cc2ac25",
                "fastqc-1_per_base_n_content_plot.yaml:md5,f43001053b544ba5585040c18859a098",
                "fastqc-1_per_base_sequence_quality_plot.yaml:md5,48276488936fd9c2e52b53a58dae0ffb",
                "fastqc-1_per_sequence_gc_content_plot_Counts.yaml:md5,8bd25680506b10fe51bc4945b1656c4c",
                "fastqc-1_per_sequence_gc_content_plot_Percentages.yaml:md5,3779a4e9c77ef8c227ea81d9888951f1",
                "fastqc-1_per_sequence_quality_scores_plot.yaml:md5,833a15c40019cb984d098baa07e221b9",
                "fastqc-1_sequence_counts_plot.yaml:md5,f0bcde34333ad21ba714a6aeb8d6a858",
                "fastqc-1_sequence_duplication_levels_plot.yaml:md5,93c7de02685a5a922b310ad94b67cf2f",
                "fastqc-status-check-heatmap.yaml:md5,45096eff6b0531cbe7ba0771e272d65d",
                "fastqc_per_base_n_content_plot.yaml:md5,ea488ff73d10d4a84b551020409db087",
                "fastqc_per_base_sequence_quality_plot.yaml:md5,73a043ea7f124e6474ac11270533de03",
                "fastqc_per_sequence_gc_content_plot_Counts.yaml:md5,c900477e0cd3a369f269bc65857377ed",
                "fastqc_per_sequence_gc_content_plot_Percentages.yaml:md5,84477f6801dcadfe0d759c2c50799b6e",
                "fastqc_per_sequence_quality_scores_plot.yaml:md5,c04ad6e6a6d61f2bbc238c61e1883317",
                "fastqc_sequence_counts_plot.yaml:md5,d9fd3b9d7b190aba26702cf73a50bd20",
                "fastqc_sequence_duplication_levels_plot.yaml:md5,73719c3f3fa610204a42cd7155d73db1",
                "fastqc_sequence_length_distribution_plot.yaml:md5,78c762e20322ee4d63f869961186c15c",
                "multiqc_bowtie2.yaml:md5,6607915ea98168dbcad83cda2ec464dc",
                "multiqc_bowtie2_bowtie2-2.yaml:md5,294c48fecbf1bdc5030550552a32497d",
                "multiqc_checkm2.yaml:md5,9b48d1b569bae456409527610b56a2a0",
                "multiqc_citations.yaml:md5,dcac7876946630b831fad3d816e8050f",
                "multiqc_fastqc.yaml:md5,f50da8bfd062b9399911ba19b34831db",
                "multiqc_fastqc_fastqc-1.yaml:md5,4c2c5a5e7b370b0668c1360b7927d0a8",
                "multiqc_general_stats.yaml:md5,8b8cbb9d6fe349084e3d2fe2ffdecda8",
                "multiqc_software_versions.yaml:md5,13a35a24875040df0018b0da426bcb65",
                "multiqc_trimmomatic.yaml:md5,d4aaff3725c28801ae0577d97c2f9b30",
                "trimmomatic_plot.yaml:md5,9557cd587ace0a93912179cb0deafe3b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-25T17:08:48.173464912"
    }
}